topdown/numbers.go

```diff
@@ -53,6 +54,23 @@ func builtinNumbersRange(bctx BuiltinContext, operands []*ast.Term, iter func(*a
 	return iter(ast.NewTerm(result))
 }
 
+func builtinRandIntn(a ast.Value) (ast.Value, error) {
+
+	switch a := a.(type) {
+
+	case ast.Number:
+		n, _ := a.Int()
+		if n <= 0 {
+			return nil, builtins.NewOperandErr(1, " should be >= 0")
+		}
+		result := rand.Intn(n)
+		return ast.InterfaceToValue(result)
+	}
+
+	return nil, builtins.NewOperandTypeErr(1, a, "int", "int64")
+}
+
 func init() {
 	RegisterBuiltinFunc(ast.NumbersRange.Name, builtinNumbersRange)
+	RegisterFunctionalBuiltin1(ast.RandIntn.Name, builtinRandIntn)
```

Comment:
That register function is deprecated. Using the other one requires changing the signature a bit, but not too much.

Reply:
I changed it but it's failing now.

Reply:
```
func builtinRandIntn(bctx BuiltinContext, args []*ast.Term, iter func(*ast.Term) error) error {
	n, err := builtins.IntOperand(args[0].Value, 1)
	if err != nil {
		return err
	}

	if n <= 0 {
		return builtins.NewOperandErr(1, " should be >= 0")
	}

	result := rand.Intn(n)
	bigResult := big.NewInt(int64(result))
	return iter(ast.NewTerm(builtins.IntToNumber(bigResult)))
}
```

Assume a single call to `iter()` is ok?